### ISSUE TITLE
Dividing a window into child windows

### ISSUE THREAD

**Issue by dgregorius** [ocornut/imgui#3228]
Version: Latest
Branch: docking

Back-ends: imgui_impl_opengl3.cpp + imgui_impl_glfw.cpp 
Compiler: VS 2019
Operating System: Windows 10

I just ran an experiment and tried to divide a parent window equally into two child windows. Here is a screenshot:

![Capture](https://user-images.githubusercontent.com/7042834/81980018-bdf02180-95e2-11ea-811a-a8701c96ca54.PNG)

And the code for this test:

ImGuiViewport* Viewport = ImGui::GetMainViewport();
ImGui::SetNextWindowPos( Viewport->Pos, ImGuiCond_Once );
ImGui::SetNextWindowSize( Viewport->Size, ImGuiCond_Once );

ImGuiWindowClass WindowClass;
WindowClass.ViewportFlagsOverrideClear = ImGuiViewportFlags_NoDecoration | ImGuiViewportFlags_NoTaskBarIcon;
ImGui::SetNextWindowClass( &WindowClass );

ImGui::PushStyleColor( ImGuiCol_WindowBg, TITLE_COLOR );
ImGui::PushStyleColor( ImGuiCol_ChildBg, ImVec4( 0.4f, 0.2f, 0.2f, 1.0f ) );
ImGui::PushStyleVar( ImGuiStyleVar_WindowPadding, ImVec2( 0, 0 ) );
if ( ImGui::Begin( "Animation", &mActive, ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoDocking ) )
{
	ImVec2 WindowPos = ImGui::GetWindowPos();
	ImVec2 WindowSize = ImGui::GetWindowSize();
	ImVec2 WindowMin = WindowPos + ImGui::GetCursorPos();
	ImVec2 WindowMax = WindowPos + WindowSize;
	ImVec2 WindowSpace = WindowMax - WindowMin;

	ImGui::BeginChild( "##Child1", ImVec2( 0, 0.5f * WindowSpace.y ) );
	ImGui::TextUnformatted( "Child1" );
	ImGui::EndChild();

	ImGui::BeginChild( "##Child2", ImVec2( 0, 0.5f * WindowSpace.y ) );
	ImGui::TextUnformatted( "Child2" );
	ImGui::EndChild();
}
ImGui::End();
ImGui::PopStyleColor( 2 );
ImGui::PopStyleVar();

Basically I am getting the window size and subtract the cursor position to account for the menu. And then I create the two child windows. The sizes add up, but I still end up with a scrollbar which means I am overshooting. What is the proper way to subdivide a window with child windows?

In the same context I would also like to have a clear border line **only at the top** of each child window. Of course you can easily achieve this by adding a line to the child windows draw list, but I am wondering if there is a better way to do this?

Thanks,
-Dirk

**Comment by ocornut** [comment]
Items are vertically spaced with `style.ItemSpacing.y`
So you can set that value to zero or calculate your sizes based on it.
